src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/WindowsForms/WindowsFormsAddItemFilter.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.WindowsForms
+{
+    /// <summary>
+    /// A filter for the Add New Item dialog that filters out Windows Forms items from non-Windows Forms projects.
+    /// </summary>
+    [ExportProjectNodeComService(typeof(IVsFilterAddProjectItemDlg))]
+    [AppliesTo("!" + ProjectCapability.WindowsForms)]
+    internal class WindowsFormsAddItemFilter : IVsFilterAddProjectItemDlg
+    {
+        public int FilterTreeItemByLocalizedName(ref Guid rguidProjectItemTemplates, string pszLocalizedName, out int pfFilter)
+        {
+            pfFilter = 0;
+            return HResult.NotImplemented;
+        }
+
+        public int FilterTreeItemByTemplateDir(ref Guid rguidProjectItemTemplates, string pszTemplateDir, out int pfFilter)
+        {
+            pfFilter = 0;
+            // Most of the templates for Windows Forms items are filtered by capabilities in the .vstemplate but there are a couple
+            // that use an older .vsz tmeplate format that doesn't support capabilities, so we filter them out here.
+            // The AppliesTo on this class ensures this code only runs for Windows Forms projects
+            if (pszTemplateDir.EndsWith("\\Windows Forms", StringComparisons.Paths))
```

Comment:
I don't follow this, this says "only run for WinForms" but the AppliesTo says "!WindowForms". How does this filter out these templates when the project is a WinForms template?

Reply:
Oops, the comment is worded incorrectly. It should say this "only runs for non-Windows Forms projects"...  but also, I forgot that we don't want the two templates for Winforms projects either, so this whole thing is wrong 🤦‍♂️